frontend/src/js/store/organizationSlice/thunks.ts

```diff
@@ -221,8 +221,12 @@ interface editTenantBody {
 export const editTenantDeviceLimit = createAsyncThunk(`${sliceName}/editDeviceLimit`, ({ newLimit, id, name }: editTenantBody, { dispatch }) => {
   return Api.put(`${tenantadmApiUrlv2}/tenants/${id}/child`, { device_limit: newLimit, name })
     .catch(err => commonErrorHandler(err, `Device Limit cannot be changed`, dispatch))
-    .then(() => Promise.resolve(dispatch(setSnackbar({ message: 'Device Limit was changed successfully', autoHideDuration: TIMEOUTS.fiveSeconds }))))
-    .then(() => dispatch(getTenants()));
+    .then(() => {
+      const tasks = [Promise.resolve(dispatch(setSnackbar('Device Limit was changed successfully')))];
+      tasks.push(dispatch(getTenants()));
+      tasks.push(dispatch(getUserOrganization()));
+      return Promise.all(tasks);
+    });
```

Comment:
```suggestion
    .then(() => {
      const tasks = [Promise.resolve(dispatch(setSnackbar('Device Limit was changed successfully'))), dispatch(getTenants()), dispatch(getUserOrganization())];
      return Promise.all(tasks);
    });
```
as the `tasks` won't get touched in any other way?
or even:
```suggestion
    .then(() => Promise.all([Promise.resolve(dispatch(setSnackbar('Device Limit was changed successfully'))), dispatch(getTenants()), dispatch(getUserOrganization())]));
```